light-curve/light_curve/light_curve_py/features/rainbow/generic.py

```diff
@@ -0,0 +1,133 @@
+    fit_and_get_errors(t, m, sigma, band, sorted=False, check=True, print_level=None, get_initial=False)
+        The same as `__call__` but also returns the parameter errors. Optionally sets the `print_level`
+        (verbosity) for Minuit fitter. If `get_initial` is True, returns the initial parameters instead
+        of fitted ones (useful for debugging)
+    """
+
+    bolometric: Union[str, BaseBolometricTerm] = dataclass_field(default="bazin", kw_only=True)
+    """Which parametric bolometric term to use"""
+    temperature: Union[str, BaseTemperatureTerm] = dataclass_field(default="sigmoid", kw_only=True)
+    """Which parametric temperature term to use"""
+
+    def __post_init__(self):
+        if not isinstance(self.bolometric, BaseBolometricTerm):
+            self.bolometric = bolometric_terms[self.bolometric]
+
+        if not isinstance(self.temperature, BaseTemperatureTerm):
+            self.temperature = temperature_terms[self.temperature]
+
+        super().__post_init__()
+
+    def _common_parameter_names(self) -> List[str]:
+        bolometric_parameters = self.bolometric.parameter_names()
+        temperature_parameters = self.temperature.parameter_names()
+        return [j for j in bolometric_parameters if j in temperature_parameters]
+
+    def _bolometric_parameter_names(self) -> List[str]:
+        bolometric_parameters = self.bolometric.parameter_names()
+        return [i for i in bolometric_parameters if i not in self._common_parameter_names()]
+
+    def _temperature_parameter_names(self) -> List[str]:
+        temperature_parameters = self.temperature.parameter_names()
+        return [i for i in temperature_parameters if i not in self._common_parameter_names()]
+
+    def bol_func(self, t, params):
+        return self.bolometric.value(t, *params[self.p.all_bol_idx])
+
+    def temp_func(self, t, params):
+        return self.temperature.value(t, *params[self.p.all_temp_idx])
+
+    def _unscale_parameters(self, params, t_scaler: Scaler, m_scaler: MultiBandScaler, scale_errors=False) -> None:
+        already_unscaled = {}
+        for term in [self.bolometric, self.temperature]:
+            for name, scaling in zip(term.parameter_names(), term.parameter_scalings()):
+                if name in already_unscaled:
+                    # Avoid un-scaling common parametres twice
+                    continue
+
+                if scaling == "time":
+                    if scale_errors:
+                        params[self.p[name]] = t_scaler.undo_scale(params[self.p[name]])
+                    else:
+                        params[self.p[name]] = t_scaler.undo_shift_scale(params[self.p[name]])
+
+                elif scaling == "timescale":
+                    params[self.p[name]] = t_scaler.undo_scale(params[self.p[name]])
+
+                elif scaling == "flux":
+                    params[self.p[name]] = m_scaler.undo_scale(params[self.p[name]])
```

Comment:
We must rise an exception for an unknown scaling

Reply:
done